uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlement.php

```diff
@@ -632,23 +683,33 @@ public static function getBreakdown($params) {
     }
 
     $leaveBalanceTypeIdOptionsGroup = self::getLeaveBalanceChangeTypeIdOptionsGroup();
+
     $results = [];
     $returnExpired = !empty($params['expired']);
     foreach($leavePeriodEntitlements as $leavePeriodEntitlement) {
       $periodEntitlementInfo = [
         'id' => $leavePeriodEntitlement->id,
         'breakdown' => []
       ];
+
       $balanceChanges = $leavePeriodEntitlement->getBreakdownBalanceChanges($returnExpired);
-      foreach($balanceChanges as $balanceChange) {
+      $balanceChangesByType = self::groupBalanceChangesByType($balanceChanges);
+
+      foreach($balanceChangesByType as $typeID => $balanceChangeGroup) {
+        $amount = array_reduce($balanceChangeGroup, function($totalAmount, $balanceChange) {
+          return $totalAmount + (float)$balanceChange->amount;
+        }, 0);
+
         $periodEntitlementInfo['breakdown'][] = [
-          'amount' => $balanceChange->amount,
-          'expiry_date' => $balanceChange->expiry_date,
-          'type' => $leaveBalanceTypeIdOptionsGroup[$balanceChange->type_id],
+          'amount' => $amount,
+          'expiry_date' => $balanceChangeGroup[0]->expiry_date,
```

Comment:
Just clarifying something. Will all balance changes of the same type of the period entitlement have the same expiry date? Is it possible for their expiry date to be different?